blocks/modal/modal.js

```diff
@@ -2,14 +2,20 @@ import { loadFragment } from '../fragment/fragment.js';
 import {
   buildBlock, decorateBlock, loadBlock, loadCSS,
 } from '../../scripts/aem.js';
+import { getSubmitBaseUrl } from '../form/constant.js';
 
 /*
   This is not a traditional block, so there is no decorate function.
   Instead, links to a /modals/ path are automatically transformed into a modal.
   Other blocks can also use the createModal() and openModal() functions.
 */
 
-export async function createModal(contentNodes) {
+let timer;
+let modelOpened = false;
```

Comment:
modelOpened and timer are doing the same thing. One of them is redundant

Reply:
Let's call this variable stopTriggers = true/false and based on that handle the triggers.

Reply:
timer is required in the case of exit intent delay, so when user exits window a timer starts, when he reenters the timer has to be stopped, 
I can set stopTimer variable as true in mouse enter but again will have to toggle this inside mouse leave event. And this would cause multiple modals to be opened every time mouse leave is triggered.